### PR TITLE
Update Azure Monitor readme.{go/python}.md with the latest specs

### PR DESCRIPTION
**Pull request by herreraj-ms**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [x] SDK of .NET (need service team to ensure code readiness)
      - [x] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [x] SDK of Go
      - [x] PowerShell
      - [x] CLI
      - [x] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [ ] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by herreraj-ms** [comment]
This PR is related to the latest specs in Azure Monitor (https://github.com/Azure/azure-rest-api-specs/pull/13852)
- Does [readme](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/monitor/resource-manager/readme.md) need a new package section?
   - Currently the tag is on package-2020-03, but there's several "only" tags (like package-2021-04-only) that seems to be missing from generating causing errors.
   - [PR](https://github.com/Azure/azure-sdk-for-net/pull/21128) SDK .NET

**Review by weidongxu-microsoft — APPROVED** [review]

**Review by ArcturusZhang — APPROVED** [review]

**Comment by herreraj-ms** [comment]
Hi, if there's something pending for merging this changes, please let me know!

**Comment by weidongxu-microsoft** [comment]
@ArcturusZhang 
go build fails.

**Comment by herreraj-ms** [comment]
> @ArcturusZhang
> go build fails.

- The problem is ErrorResponse Model (DuplicateModelCollsion).
- The specs in 2021-04-01 (Data Collection) uses v2 ErrorResponse reference (common-types/resource-management/v2/types.json#/definitions/ErrorResponse)
- All other specs define ErrorResponse in-line with two properties code & message
- Special case is "stable/2018-03-01/metricAlert_API.json" which defines ErrorResponse in-line but then extend it to ErrorResponseCommon for mimic common-types/resource-management/v1/types.json#/definitions/ErrorResponse

This PR is only to make a new package, but the error should be fixed on Autorest command, no?

File: specification/monitor/resource-manager/readme.md [hunk 1]
```diff
@@ -34,7 +34,38 @@ title: MonitorManagementClient
...
+input-file:
+- Microsoft.Insights/stable/2015-04-01/autoscale_API.json
+- Microsoft.Insights/stable/2015-04-01/operations_API.json
+- Microsoft.Insights/stable/2016-03-01/alertRulesIncidents_API.json
+- Microsoft.Insights/stable/2016-03-01/alertRules_API.json
+- Microsoft.Insights/stable/2016-03-01/logProfiles_API.json
+- Microsoft.Insights/preview/2017-05-01-preview/diagnosticsSettings_API.json
+- Microsoft.Insights/preview/2017-05-01-preview/diagnosticsSettingsCategories_API.json
+- Microsoft.Insights/stable/2019-06-01/actionGroups_API.json
+- Microsoft.Insights/stable/2015-04-01/activityLogs_API.json
```
Comment (ArcturusZhang) [inline specification/monitor/resource-manager/readme.md:54]: Per request here: https://github.com/Azure/azure-sdk-for-go/issues/14704
Could this be updated to a newer api-version?

**Review by ArcturusZhang — CHANGES_REQUESTED** [review]

File: specification/monitor/resource-manager/readme.md (hunk 1 above)
Comment (herreraj-ms) [reply specification/monitor/resource-manager/readme.md:54]: The only spec in [2017-03-01-preview](https://github.com/Azure/azure-rest-api-specs/tree/master/specification/monitor/resource-manager/Microsoft.Insights/preview/2017-03-01-preview) is activityLogAlerts_API.json.

The latest for activityLogs_API.json is 2015-04-01.
https://docs.microsoft.com/en-us/rest/api/monitor/activitylogs

I don't see other version of Activity Logs.
Or, Am I misunderstanding?

File: specification/monitor/resource-manager/readme.md [hunk 2]
```diff
@@ -34,7 +34,38 @@ title: MonitorManagementClient
...
+- Microsoft.Insights/stable/2015-04-01/tenantActivityLogs_API.json
+- Microsoft.Insights/stable/2018-01-01/metricDefinitions_API.json
+- Microsoft.Insights/stable/2018-01-01/metrics_API.json
+- Microsoft.Insights/stable/2019-03-01/metricBaselines_API.json
+- Microsoft.Insights/stable/2018-03-01/metricAlert_API.json
+- Microsoft.Insights/stable/2018-04-16/scheduledQueryRule_API.json
+- Microsoft.Insights/preview/2017-12-01-preview/metricNamespaces_API.json
+- Microsoft.Insights/preview/2018-11-27-preview/vmInsightsOnboarding_API.json
+- Microsoft.Insights/preview/2019-10-17-preview/privateLinkScopes_API.json
+- Microsoft.Insights/stable/2017-04-01/activityLogAlerts_API.json
```
Comment (ArcturusZhang) [inline specification/monitor/resource-manager/readme.md:65]: Per the error message in [this issue](https://github.com/Azure/azure-sdk-for-go/issues/14704), the available api-version is listed as
```
The supported api-versions are '2014-04-01,2014-11-01,2015-04-01,2016-09-01-preview,2017-03-01-preview'
```
2017-04-01 is not one of them. Could we double confirm this is OK?

**Review by ArcturusZhang — CHANGES_REQUESTED** [review]

File: specification/monitor/resource-manager/readme.md (hunk 1 above)
Comment (ArcturusZhang) [reply specification/monitor/resource-manager/readme.md:54]: Sure, I am fine with this. Please take a look at the comment below

File: specification/monitor/resource-manager/readme.md (hunk 2 above)
Comment (herreraj-ms) [reply specification/monitor/resource-manager/readme.md:65]: activityLogAlerts_API.json & activityLogs_API.json are two different specs

**Comment by ArcturusZhang** [comment]
> > @ArcturusZhang
> > go build fails.
> 
> * The problem is ErrorResponse Model (DuplicateModelCollsion).
> * The specs in 2021-04-01 (Data Collection) uses v2 ErrorResponse reference (common-types/resource-management/v2/types.json#/definitions/ErrorResponse)
> * All other specs define ErrorResponse in-line with two properties code & message
> * Special case is "stable/2018-03-01/metricAlert_API.json" which defines ErrorResponse in-line but then extend it to ErrorResponseCommon for mimic common-types/resource-management/v1/types.json#/definitions/ErrorResponse
> 
> This PR is only to make a new package, but the error should be fixed on Autorest command, no?

Yes, we need to fix this in order to generate the SDK - the best solution is to change every swagger file to use the ErrorResponse defined in the common-types. I have done that once when exactly the same situation happens to subscriptionIdParameter.

Could we fix this?

File: specification/monitor/resource-manager/readme.md (hunk 2 above)
Comment (ArcturusZhang) [reply specification/monitor/resource-manager/readme.md:65]: I am as confused as the customer right now. Is it possible that you could go to [this issue](https://github.com/Azure/azure-sdk-for-go/issues/14704) and explain this?

Comment (herreraj-ms) [reply specification/monitor/resource-manager/readme.md:65]: Using rest API directly (https://docs.microsoft.com/en-us/rest/api/monitor/activity-logs/list) I was able to see the error message.
ARMs manifest I do see the versions, but no spec file. That why GO client do not reflect the latest API.
I'll contact the RP owner to understand why there's no spec file(s)

**Comment by herreraj-ms** [comment]
> > > @ArcturusZhang
> > > go build fails.
> > 
> > 
> > 
> > * The problem is ErrorResponse Model (DuplicateModelCollsion).
> > * The specs in 2021-04-01 (Data Collection) uses v2 ErrorResponse reference (common-types/resource-management/v2/types.json#/definitions/ErrorResponse)
> > * All other specs define ErrorResponse in-line with two properties code & message
> > * Special case is "stable/2018-03-01/metricAlert_API.json" which defines ErrorResponse in-line but then extend it to ErrorResponseCommon for mimic common-types/resource-management/v1/types.json#/definitions/ErrorResponse
> > 
> > This PR is only to make a new package, but the error should be fixed on Autorest command, no?
> 
> Yes, we need to fix this in order to generate the SDK - the best solution is to change every swagger file to use the ErrorResponse defined in the common-types. I have done that once when exactly the same situation happens to subscriptionIdParameter.
> 
> Could we fix this?

Can you provide your alias to continue in an internal mail thread?

**Comment by herreraj-ms** [comment]
@ArcturusZhang I check-in changes as directives and references (definitions/parameters) for data collection rules specs, for fixing the checks.

Regarding the GO, I cannot figure out this error "the output-folder of a preview package 'package-2021-04' must be under the `preview` subdirectory"

Could you help with this one?

**Comment by herreraj-ms** [comment]
@weidongxu-microsoft I'm fairly sure that the check "Swagger LintDiff" will go away if I add the directive used for .NET/Go.

Which yaml tag can I use targeting azure-validator?
This is the cmd from the pipeline: autorest --validation --azure-validator --semantic-validator=false --model-validator=false --message-format=json --openapi-type=arm --use=@microsoft.azure/classic-openapi-validator@1.1.8 --use=@microsoft.azure/openapi-validator@1.10.0 --tag=package-2021-04 specification/monitor/resource-manager/readme.md


I want an expression in readme.md like (yaml $(go) || $(csharp) || $(azure-validator)):
``` yaml $(go) || $(csharp) || $(azure-validator)
directive:
- from: activityLogAlerts_API.json
  where: $.definitions
  transform: delete $["Resource"]
  reason: Missing kind, etag; Generation will take the definition from scheduledQueryRule_API.json which includes kind & etag
```

File: specification/monitor/resource-manager/readme.go.md [hunk 3]
```diff
@@ -102,5 +107,5 @@ These settings apply only when `--tag=package-2021-04 --go` is specified on the
 Please also specify `--go-sdk-folder=<path to the root directory of your azure-sdk-for-go clone>`.
 
 ``` yaml $(tag) == 'package-2021-04' && $(go)
-output-folder: $(go-sdk-folder)/services/monitor/mgmt/2021-04-01/$(namespace)
+output-folder: $(go-sdk-folder)/services/monitor/mgmt/2021-04-01-preview/$(namespace)
```
Comment (ArcturusZhang) [inline specification/monitor/resource-manager/readme.go.md:110]:
``` suggestion
output-folder: $(go-sdk-folder)/services/preview/monitor/mgmt/2021-04-01-preview/$(namespace)
```

**Review by ArcturusZhang — CHANGES_REQUESTED** [review]

File: specification/monitor/resource-manager/readme.go.md (hunk 3 above)
Comment (herreraj-ms) [reply specification/monitor/resource-manager/readme.go.md]: thanks! fixed.

**Comment by herreraj-ms** [comment]
@weidongxu-microsoft could you look at this one? 
I want to know if there are fixes pending on my side.

**Comment by weidongxu-microsoft** [comment]
@herreraj-ms Please check with ArcturusZhang as he requires changes. Also no one can merge if LintDiff and BreakChanges CI fails.

**Comment by herreraj-ms** [comment]
> @herreraj-ms Please check with ArcturusZhang as he requires changes. Also no one can merge if LintDiff and BreakingChanges CI fails.

@weidongxu-microsoft Both checks have been fixed!

@ArcturusZhang GO build is successful, is there other expected fix to be done?

**Comment by weidongxu-microsoft** [comment]
@ArcturusZhang Is there any change you requrie?

**Comment by herreraj-ms** [comment]
@ArcturusZhang ping

**Review by ArcturusZhang — APPROVED** [review]

**Comment by ArcturusZhang** [comment]
Hi @herreraj-ms if you could fix the merge conflicts, we are good to merge! thank you

**Comment by herreraj-ms** [comment]
> Hi @herreraj-ms if you could fix the merge conflicts, we are good to merge! thank you

Done!

**Pull request merged**